plugins/mimecast/komand_mimecast/connection/connection.py

```diff
@@ -43,9 +43,24 @@ def test_task(self):
         self.logger.info("Running a connection test to Mimecast")
         try:
             _, _, _ = self.client.get_siem_logs("")
-            self.logger.info("The connection test to Mimecast was successful")
-            return {"success": True}
+            message = "The connection test to Mimecast was successful"
+            self.logger.info(message)
+            return {"success": True}, message
         except ApiClientException as error:
-            self.logger.info("The connection test to Mimecast has failed")
+
+            return_message = ""
+
+            failed_message = "The connection test to Mimecast has failed"
+            self.logger.info(failed_message)
+            return_message += f"{failed_message}\n"
+
+            cause_message = f"This failure was caused by: {error.cause}"
```

Comment:
These look fine to me. Do we have examples of the cause and assistance outside of the standardised error messages? I know sometimes the assistance messages don't provide a direct solution only guidance.

It's less explicit but we could use `cause:` and `assistance` if there's more variability in the values.

Reply:
The main list of the cause and assistance come form here - https://github.com/rapid7/insightconnect-plugins/blob/f2ccfe5052e5f77c9ba09f46428822e05bbbde3b/plugins/mimecast/komand_mimecast/util/api.py#L316 

This is still open to debate if there are better solutions this was just me trying to reuse as much of what was already there to get something of the ground

Reply:
Another minor, should the cause be in nested quotes? `"This failure was caused by 'Incorrect App ID supplied'"` ?

Reply:
This has been updated to include single `''` around the cause field